src/schedule/zephyr.c

```diff
@@ -63,8 +63,11 @@ static void idc_handler(struct k_p4wq_work *work)
 
 	SOC_DCACHE_INVALIDATE(msg, sizeof(*msg));
 
-	if (msg->size == sizeof(int))
-		assert(!memcpy_s(&payload, sizeof(payload), msg->payload, msg->size));
+	if (msg->size == sizeof(int)) {
+		const int idc_handler_memcpy_err __unused =
+			memcpy_s(&payload, sizeof(payload), msg->payload, msg->size);
+		assert(!idc_handler_memcpy_err);
```

Comment:
Totally a style thing, but I'd view `if (memcpy(...)) { assert(false); }` as more readable than creating a variable you need to annotate with __unused because of a compiler warning.  Or even adding a `checked_memcpy()` variant that will implicitly assert on an error?  Meh, probably not worth the effort.

Reply:
Thanks, I'd love to get rid of the `__unused` and this looks like a great way to do that but I think some `assert(some_readable_condition)` macros are able to print `some_readable_condition` somewhere sometimes when they fail depending on which debugguer/trace/crashdump you use... Is it the case with Zephyr's or SOF's assert? If neither then I totally want to switch to your suggestion as is. If at least one of them does then I'm not sure what's best.

Reply:
Indeed, Zephyr's __ASSERT() will print the text of the expression, and it even has a printf-style format string and arguments you can use.  And this would defeat that, you'd just get the file/line info.  Which is frankly all I want 94% of the time.

I guess mostly I don't actually want to see the assertions when I'm reading code. :)

Reply:
Change of mind: I found that:
- `__unused` is used in the official Zephyr documentation https://docs.zephyrproject.org/apidoc/latest/group__log__api.html#details
- `#define assert(expr) __ASSERT(expr, #expr)` prints the asserted expression, which is nice.

So I went one step further and added one commit that makes the Zephyr build compatible with Zephyr's `__ASSERT` (without actually switching)

Reply:
is there any reason why we cant use the native zephyr ```ASSERT()``` and redefine if for xtos usage ?

Reply:
> is there any reason why we cant use the native zephyr ASSERT() and redefine if for xtos usage ?

That should be possible but there are other asserts elsewhere (in non-Zephyr code) that this PR does not touch yet and it is already pretty big.